scalafix-core/shared/src/main/scala/scalafix/internal/rule/ProcedureSyntax.scala

```diff
@@ -1,40 +1,37 @@
 package scalafix.internal.rule
 
-import scala.collection.immutable.Seq
 import scala.meta._
 import scala.meta.tokens.Token.LeftBrace
-import scala.meta.tokens.Token.RightParen
-import scala.meta.tokens.Token.RightBracket
 import scalafix.Patch
 import scalafix.rule.Rule
 import scalafix.rule.RuleCtx
+import scalafix.rule.RuleName
+import scalafix.util.Trivia
+
+case object ProcedureSyntax
+    extends Rule(
+      RuleName("ProcedureSyntax").withDeprecatedName(
+        "ExplicitUnit",
+        "Use ProcedureSyntax instead",
+        "0.6.0"
+      )
+    ) {
 
-case object ProcedureSyntax extends Rule("ProcedureSyntax") {
   override def description: String =
     "Rewrite that inserts explicit : Unit = for soon-to-be-deprecated procedure syntax def foo { ... }"
 
   override def fix(ctx: RuleCtx): Patch = {
-    val patches: Seq[Patch] = ctx.tree.collect {
+    ctx.tree.collect {
+      case t: Decl.Def if t.decltpe.tokens.isEmpty =>
+        ctx.addRight(t.tokens.last, s": Unit").atomic
       case t: Defn.Def
           if t.decltpe.exists(_.tokens.isEmpty) &&
             t.body.tokens.head.is[LeftBrace] =>
-        val bodyStart = t.body.tokens.head
-        val defEnd = (for {
-          lastParmList <- t.paramss.lastOption
-          lastParam <- lastParmList.lastOption
-        } yield lastParam.tokens.last).getOrElse(t.name.tokens.last)
-        val tokens =
-          ctx.tokenList.slice(ctx.tokenList.next(defEnd), bodyStart)
-        val closingBracketOrParen =
-          tokens
-            .find(_.is[RightBracket])
-            .getOrElse(
-              tokens
-                .find(_.is[RightParen])
-                .getOrElse(defEnd)
-            )
-        ctx.addRight(closingBracketOrParen, s": Unit =").atomic
-    }
-    patches.asPatch
+        val fixed = for {
+          bodyStart <- t.body.tokens.headOption
+          toAdd <- ctx.tokenList.leading(bodyStart).find(!_.is[Trivia])
+        } yield ctx.addRight(toAdd, s": Unit =").atomic
```

Comment:
I am not familiar with these tokens, so this change looks magic to me, but since the diff on test data looks fine so LGTM.

Reply:
The trick is `tokenList.leading` which returns a view of the tokens in reverse order starting from the `{` of the method body. `Trivia` is a category for whitespace and comment tokens.